java/flight/src/main/java/org/apache/arrow/flight/FlightClient.java

```diff
@@ -86,35 +86,83 @@ public FlightClient(BufferAllocator incomingAllocator, Location location) {
 
   /**
    * Get a list of available flights.
+   *
    * @param criteria Criteria for selecting flights
    * @return FlightInfo Iterable
    */
   public Iterable<FlightInfo> listFlights(Criteria criteria) {
-    return ImmutableList.copyOf(blockingStub.listFlights(criteria.asCriteria()))
+    return listFlights(null, criteria);
+  }
+
+  /**
+   * Get a list of available flights.
+   *
+   * @param options RPC-layer hints for the call. May be null for default options.
+   * @param criteria Criteria for selecting flights
+   * @return FlightInfo Iterable
+   */
+  public Iterable<FlightInfo> listFlights(CallOptions options, Criteria criteria) {
```

Comment:
I've been thinking about this further. I'm not a huge fan of the double methods. What do you think about changing this to using vararg to manage both:

e.g. listFlights(Criteria c, CallOption...options)

Then we can simply introduce new CallOption items in the future. It means a user can call without any options or a selection of options as needed without introducing multiple apis.

Thoughts?